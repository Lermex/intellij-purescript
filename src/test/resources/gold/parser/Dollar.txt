Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PSASTWrapperElement(TypeAnnotationName)
          PsiElement(identifier)('applyFn')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('b')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(()('(')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(GenericIdentifier)
                            PsiElement(identifier)('b')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PsiElement(->)('->')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('b')
                          PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('applyFn')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('f')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('x')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('f')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(FixityDeclaration)
        PSASTWrapperElement(Fixity)
          PsiElement(infixr)('infixr')
          PsiElement(whitespace)(' ')
          PsiElement(natural)('1000')
          PsiElement(whitespace)(' ')
        PsiElement(identifier)('applyFn')
        PsiElement(whitespace)(' ')
        PsiElement(as)('as')
        PsiElement(whitespace)(' ')
        PsiElement(operator)('$')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('id')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('x')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test1')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('x')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)(' ')
            PSASTWrapperElement(IdentInfix)
              PSASTWrapperElement(Qualified)
                PsiElement(operator)('$')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Value)
              PSASTWrapperElement(PrefixValue)
                PSASTWrapperElement(Var)
                  PSASTWrapperElement(Qualified)
                    PsiElement(identifier)('id')
                    PsiElement(whitespace)(' ')
              PSASTWrapperElement(IdentInfix)
                PSASTWrapperElement(Qualified)
                  PsiElement(operator)('$')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('id')
                      PsiElement(whitespace)(' ')
                PSASTWrapperElement(IdentInfix)
                  PSASTWrapperElement(Qualified)
                    PsiElement(operator)('$')
                    PsiElement(whitespace)(' ')
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('x')
                        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test2')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('x')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('x')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')