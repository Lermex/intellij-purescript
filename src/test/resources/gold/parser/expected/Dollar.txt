Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('applyFn')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('a')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('b')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(()('(')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeVar)
                            PsiElement(identifier)('b')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PsiElement(->)('->')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('b')
                          PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('applyFn')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('f')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('x')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('f')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(FixityDeclaration)
        PSASTWrapperElement(Fixity)
          PsiElement(infixr)('infixr')
          PsiElement(whitespace)(' ')
          PsiElement(natural)('1000')
          PsiElement(whitespace)(' ')
        PsiElement(identifier)('applyFn')
        PsiElement(whitespace)(' ')
        PsiElement(as)('as')
        PsiElement(whitespace)(' ')
        PsiElement(operator)('$')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('id')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('x')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test1')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('x')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)(' ')
            PSASTWrapperElement(IdentInfix)
              PSASTWrapperElement(Qualified)
                PsiElement(operator)('$')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Value)
              PSASTWrapperElement(PrefixValue)
                PSASTWrapperElement(Var)
                  PSASTWrapperElement(Qualified)
                    PsiElement(identifier)('id')
                    PsiElement(whitespace)(' ')
              PSASTWrapperElement(IdentInfix)
                PSASTWrapperElement(Qualified)
                  PsiElement(operator)('$')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('id')
                      PsiElement(whitespace)(' ')
                PSASTWrapperElement(IdentInfix)
                  PSASTWrapperElement(Qualified)
                    PsiElement(operator)('$')
                    PsiElement(whitespace)(' ')
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('x')
                        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test2')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('x')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('id')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('x')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')